bond/bond.go

```diff
@@ -228,6 +229,12 @@ func setLinksinNetNs(bondConf *bondingConfig, nspath string, releaseLinks bool)
 		return fmt.Errorf("Bonding requires at least two links, we have %+v", len(linkNames))
 	}
 
+	if releaseLinks {
+		if err := curnetNs.Set(); err != nil {
```

Comment:
perhaps instead of using Set (here and the one above L197), you can try to use Do. Based on the value of releaseLinks, you can do something similar to what you do for addLinksInNS in the JustBeforeEach block.

What do you think?